src/tainting/Taint.ml

```diff
@@ -108,16 +108,26 @@ let rec _show_call_trace show_thing = function
 (* Signatures *)
 (*****************************************************************************)
 
-type arg_pos = string * int [@@deriving show, compare]
-type arg = { pos : arg_pos; offset : IL.name list } [@@deriving show]
+type arg_pos = { name : string; index : int } [@@deriving show, compare]
+type arg_base = Bthis | Barg of arg_pos [@@deriving show, compare]
+type arg = { base : arg_base; offset : IL.name list } [@@deriving show]
 
-let _show_arg { pos = s, i; offset = os } =
+let _show_pos { name = s; index = i } = Printf.sprintf "arg(%s@%d)" s i
+
+let _show_base base =
```

Comment:
why the _prefix? And why not use deriving show instead?

Reply:
There is a `deriving show` that is useful to display a small amount of info, but usually it's too verbose, so the `_show` functions are better to everyday's debugging. Happy to change the prefix to something else if there is a good suggestion.

Reply:
maybe debug_ then? You can also annotate the types and adjust what deriving show is doing by default if the default is too verbose for you.